R/allele_qc.R

```diff
@@ -1,38 +1,86 @@
+    data$pos <- as.integer(data$pos)
+    return(data)
+  }
+
+  # Check if id1 is in the first vector format
+  if (any(grepl(":", variant_id[1])) && any(grepl("_", variant_id[1]))) {
+    return(create_dataframe(variant_id, "colon_underscore"))
+  }
+
+  # Check if id1 is in the second vector format
+  if (all(grepl(":", variant_id[1]))) {
+    return(create_dataframe(variant_id, ":"))
+  }
+  # If none of the conditions are met, stop and print an error
+  stop("Input does not match any expected format. Please provide a valid data frame or a character vector in the specified formats.")
+}
+#' Match alleles between target_variants and ref_variants
+#'
 #' Match by ("chrom", "A1", "A2" and "pos"), accounting for possible
 #' strand flips and major/minor allele flips (opposite effects and zscores).
-#' 
-#' @param sumstats A data frame with columns "chrom", "pos", "A1", "A2", "beta" and "z". 
-#' @param info_snp A data frame with columns "chrom", "pos", "A1" and "A2".
-#' @param match.min.prop  Minimum proportion of variants in the smallest data
-#' to be matched, otherwise stops with an error. Default is 20\%.
-#' @param remove_dups Whether to remove duplicates, default is True.
-#' @param flip Whether the alleles must be flipped: A <--> T & C <--> G, in which case corresponding `$beta` and `$z` are multiplied by -1
-#' Default is `TRUE`.
-#' @param remove Whether to remove strand SNPs (if any)
 #'
-#' @return A single data frame with matched variants. Values in columns "beta" and "z"
-#'   are multiplied by -1 for variants with alleles reversed
+#' @param target_variants A data frame with columns "chrom", "pos", "A1", "A2" or strings in the format of "chr:pos:A1:A2"/"chr:pos_A1_A2".
+#' @param ref_variants A data frame with columns "chrom", "pos", "A1", "A2" or strings in the format of "chr:pos:A1:A2"/"chr:pos_A1_A2".
+#' @param target_data A data frame on which QC procedures will be applied..
+#' @param col_to_flip The name of the column in target_data where flips are to be applied.
+#' @param match_min_prop Minimum proportion of variants in the smallest data
+#'   to be matched, otherwise stops with an error. Default is 20%.
+#' @param remove_dups Whether to remove duplicates, default is TRUE.
+#' @param flip Whether the alleles must be flipped: A <--> T & C <--> G, in which case 
+#'   corresponding `col_to_flip` are multiplied by -1. Default is `TRUE`.
+#' @param remove Whether to remove strand SNPs (if any). Default is `TRUE`.
+#' @return A single data frame with matched variants.
+#' @importFrom dplyr mutate filter
 #' @importFrom vctrs vec_duplicate_detect
-#' @importFrom Rlab message2
 #' @export
-#'
-allele_qc = function(sumstats,info_snp,match.min.prop,remove_dups,flip,remove) {
-  sumstats  =  as.data.frame(sumstats)
-  info_snp  =  as.data.frame(info_snp)
+allele_qc = function(target_variants, ref_variants, target_data, col_to_flip, match.min.prop, remove_dups,flip,remove){
```

Comment:
Please provide default values to `match.min.prop=0.2, remove_dups=T,remove=T,flip=T`.